softlayer/data_source_softlayer_vlan.go

```diff
@@ -0,0 +1,74 @@
+package softlayer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+)
+
+func dataSourceSoftLayerVlan() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSoftLayerVlanRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"number", "router_hostname"},
+			},
+
+			"number": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
+
+			"router_hostname": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
```

Comment:
Not sure @minsikl, will secondary_router_hostname ever be something we need to add to the vlan resource or its corresponding data source? Hoping we can simplify down to router_hostname.

About the other fields. I can work on adding subnets to the data source, if that will be useful for informing other resources. The others, I feel we should wait until there is a need to add it.